src/app/users/users.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { RegisterService } from "../services/users/users.service";
+import { first } from "rxjs/operators";
+
+@Component({
+  selector: 'app-users',
+  templateUrl: './users.component.html',
+  styleUrls: ['./users.component.css']
+})
+export class UsersComponent implements OnInit {
+  registerForm: FormGroup;
+  show_error = false;
+  msj_error = null;
+
+  constructor(
+    private fb: FormBuilder,
+    private registerService: RegisterService
+  ) {
+  }
+
+  ngOnInit() {
+    this.registerForm = this.fb.group({
+      username: ['', Validators.required],
+      email: ['', Validators.required],
+      password1: ['', Validators.required],
+      password2: ['', Validators.required],
```

Comment:
Creo que aquí puede ir lo de la validación de las contraseñas iguales: https://stackoverflow.com/questions/51605737/confirm-password-validation-in-angular-6

Reply:
Validaciones necesarias para contraseña 
```sh
"password1": [
        "This password is too short. It must contain at least 8 characters.",
        "This password is too common.",
        "This password is entirely numeric."
    ]
```

Reply:
Si las podemos hacer antes desde el frontend, me parece mejor